src/main/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterBase.java

```diff
@@ -0,0 +1,12 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import java.io.PrintWriter;
+
+public abstract class NamePrinterBase implements NamePrinter {
+    private String liTagPattern = "<li>%s</li>";
+
+    public void print(PrintWriter writer) {
+        writer.println(String.format(liTagPattern, this.getName()));
```

Comment:
why we need to format String and why we need to use PrintWriter?
you can use String with string.replace instead.

Reply:
Or consider using StringBuilder ?